plugins/woocommerce/includes/export/abstract-wc-csv-exporter.php

```diff
@@ -372,6 +375,11 @@ public function escape_data( $data ) {
 		// 0x0d: Carriage Return (\r)
 		$active_content_triggers = array( '=', '+', '-', '@', chr( 0x09 ), chr( 0x0d ) );
 
+		// Don't escape pure numeric values since they cannot form a valid formula to be injected.
+		if ( is_int( $data ) || is_float( $data ) ) {
```

Comment:
`is_int` and `is_float` will fail for string values. Depending on how the data is stored, there is a good chance that numeric values are `string` here. `is_numeric` may be better as it checks for numeric values in strings as well as `int`/`float`

The first example in the phpdocs seems to allow everything we'd want to accept while refusing anything problematic.

Reply:
Manually testing seems to show that the data we care about (Product data, Analytics data) is stored as one of these. I'm happy to err on the side of caution with this PR and only allow int/float since it solves our immediate problem and remains "air-tight". 

Also, this is what we're doing with JS-based CSV export in #52727, so we should see consistent output.

If we do see escaped string-numeric values becoming an issue in the future, we can make this less strict, or perhaps prepare the data as int/float prior to escaping.